src/main/resources/se/simonsoft/cms/indexing/properties/subject-scheme-props.xsl

```diff
@@ -0,0 +1,75 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+
+    Copyright (C) 2009-2017 Simonsoft Nordic AB
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+            http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+-->
+<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
+                xmlns:xs="http://www.w3.org/2001/XMLSchema"
+                xmlns:map="http://www.w3.org/2005/xpath-functions/map"
+                exclude-result-prefixes="xs map"
+                version="3.0">
+
+    <xsl:param name="properties" as="map(xs:string, xs:string)"/>
+
+    <xsl:output method="xml" indent="yes"/>
+
+    <xsl:template match="subjectScheme">
+
+        <xsl:variable name="subjects" select="//enumerationdef[elementdef[@name='cms:property']][subjectdef[@keyref]]"/>
+
+        <xsl:variable name="titles" select="//subjectdef[subjectdef[topicmeta[navtitle]]]"/>
+
+        <xsl:element name="doc">
+
+            <xsl:for-each select="map:keys($properties)">
+
+                <xsl:variable name="property-name" select="." as="xs:string"/>
+                <xsl:variable name="property-name-nocolon" select="translate($property-name, ':', '.')"/>
+                <xsl:variable name="property-values" select="tokenize(map:get($properties, $property-name), ' ')" as="xs:string+"/>
+
+                <xsl:variable name="subject-key" select="$subjects[attributedef[@name=$property-name]]/subjectdef/@keyref"/>
+
+                <xsl:for-each select="$property-values">
+
+                    <xsl:variable name="property-value" select="." as="xs:string"/>
+                    <xsl:variable name="property-value-title" select="$titles[@keys=$subject-key]/subjectdef[@keys=$property-value]/topicmeta/navtitle"/>
```

Comment:
What happens if the same key exists in different subjectdef trees? I don't like the "titles" variable.

Reply:
I suggest you add another subjectdef to the test data (with an existing key) to test and get test coverage going forward.

Reply:
I'd be happy to replace it with a smarter way. What's the proper way to do it?

Reply:
You mean a duplicate at this level kind of like this one?

```xml
        <subjectdef keys="productfamily">
            <topicmeta>
                <shortdesc>Product Family</shortdesc>
            </topicmeta>
            <subjectdef keys="Tempo_T_6">
                <topicmeta>
                    <navtitle>Another Tempo T 6</navtitle>
                </topicmeta>
            </subjectdef>
        </subjectdef>
        <subjectdef keys="productname">
            <subjectdef keys="Tempo_T_6">
                <topicmeta>
                    <navtitle>Tempo T 6</navtitle>
                </topicmeta>
            </subjectdef>
        </subjectdef>
```

Reply:
Since `$titles` selects the parent subjectdef, I think it doesn't confuse the duplicate keys at this level. Once I understand your point better I can extend the test coverage for this or replace the titles with a better alternative.

Reply:
Yes, that is the test I meant.

Ok, now I understand line 48, it will probably produce correct results for the first sublevel of subjectdef.
 - It will fail to find nested subjectdef, I think.
 - It is pointless to store all subjectdef in $titles because it will only use the top subjectdef.
 
 You can remove $titles and make a single xpath on line 48.

Reply:
Try on line 48:
```
"//subjectdef[@keys=$subject-key]/descendant::subjectdef[@keys=$property-value]/topicmeta/navtitle"
```

There is also the option to use xsl:key which is likely the fastest, skip for now.

Reply:
```
Error at char 0 in xsl:variable/@select on line 48 column 171 
  XPTY0020  Leading '/' selects nothing: the context item is not a node
```